### PR TITLE
Nikhil_Assignment_20_Done_(5+5+10+5=)25Min

### PR DESCRIPTION
**Pull request by NikhilMJun20**
->Completed and pushed Assignment # 20.
->Please review and comment.

->PFB screenshots of the outputs of the programs:-

**1**

**2**

**3**

**4**

### CODE REVIEW

File: src/nikhilMJun20/ArmstrongNumber.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		if(isArm == sum)
+			System.out.println("\n"+isArm+" is an Armstrong Number.");
+		else
+			System.out.println("\n"+isArm+" is NOT an Armstrong Number.");
+	}
+	
+	public static void main(String[] args) {
+		//Other Armstrong Numbers : 370, 371, 407
+		int num = 153;
+
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/ArmstrongNumber.java]: -Remove extra space
-Otherwise logic looks good to me

File: src/nikhilMJun20/DuplicateFinder.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+			for(int innerIndex=index+1; innerIndex<intArray.length; innerIndex++) {
+				if(intArray[index] == intArray[innerIndex])
+					System.out.println("Duplicate Array Element is : "+intArray[index]);
+			}
+		}
+	}
+	
+	public static void main(String[] args) {
+		int[] intArray = {0, 3, 1, 2, 3};
+
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/DuplicateFinder.java]: --Remove extra space

File: src/nikhilMJun20/MissingNumberFromArray.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		
+		System.out.println("Given Array : {1, 2, 3, 4, 6, 7, 8, 9, 10}");
+		for(int index=0; index<intArray.length; index++) {
+			arraySum += intArray[index];
+		}
+		
+		for(int num=1; num<=10; num++) {
+			originalSum += num;
+		}
+
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/MissingNumberFromArray.java:19]: --Remove extra space

File: src/nikhilMJun20/MissingNumberFromArray.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		for(int num=1; num<=10; num++) {
+			originalSum += num;
+		}
+		
+		System.out.println("\nMissing Number from 1 to 10 is : "+(originalSum-arraySum));
+	}
+	
+	public static void main(String[] args) {
+		int[] intArray = {1, 2, 3, 4, 6, 7, 8, 9, 10};
+
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/MissingNumberFromArray.java]: Remove extra space

File: src/nikhilMJun20/MissingNumberFromArray.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+public class MissingNumberFromArray {
+	void findMissing(int[] intArray) {
+		int originalSum=0, arraySum=0;
+		
+		System.out.println("Given Array : {1, 2, 3, 4, 6, 7, 8, 9, 10}");
+		for(int index=0; index<intArray.length; index++) {
+			arraySum += intArray[index];
+		}
+		
+		for(int num=1; num<=10; num++) {
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/MissingNumberFromArray.java:16]: --This logic also looks good to me.
-- You can also make this work using only one for loop.

File: src/nikhilMJun20/ReverseInteger.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+*/
+
+package nikhilMJun20;
+
+public class ReverseInteger {
+	void revInteger(int num) {
+		int originalNum = num, revInt=0;
+		
+		while(num>0) {
+			revInt = 10*revInt+(num % 10);
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/ReverseInteger.java:14]: --You can directly write int revInt = 10*revInt+(num % 10);
-- Code logic looks good to me.

**Review by VaishnaviVJun20 — COMMENTED** [review]
-Please work on given comments.

File: src/nikhilMJun20/ReverseInteger.java (hunk 6 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/ReverseInteger.java:14]: ->If I declare/write revInt like "int revInt = 10*revInt+(num % 10);" as suggested by you, its scope will be limited to while loop only and I will not be able to use it in printing statement i.e. outside while().

File: src/nikhilMJun20/MissingNumberFromArray.java (hunk 4 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/MissingNumberFromArray.java]: ->Implemented the suggestion.

File: src/nikhilMJun20/MissingNumberFromArray.java (hunk 3 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/MissingNumberFromArray.java:19]: ->Implemented the suggestion.

File: src/nikhilMJun20/DuplicateFinder.java (hunk 2 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/DuplicateFinder.java]: ->Implemented the suggestion.

File: src/nikhilMJun20/MissingNumberFromArray.java (hunk 5 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/MissingNumberFromArray.java:16]: ->This was exactly for 1 to 10 hard coded. So, I did not use n(n+1)/2 formula.

Comment (VaishnaviVJun20) [reply src/nikhilMJun20/MissingNumberFromArray.java:16]: -- Sure then code looks good to me

Comment (VaishnaviVJun20) [reply src/nikhilMJun20/MissingNumberFromArray.java:16]: --okay good then this logic also works

**Review by VaishnaviVJun20 — APPROVED** [review]
--code looks good to me
--Variable and method names makes sense
--logic is also clean
-- Good work! Keep it up

**Comment by NikhilMJun20** [comment]
Thank you, Vaishnavi!

File: src/nikhilMJun20/MissingNumberFromArray.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+			originalSum += num;
```
Comment (KrishnaKTechnocredits) [inline src/nikhilMJun20/MissingNumberFromArray.java:17]: For originalSum you can use math operation as well. 
void findMissingNumber2(int[] array) {
		int al = array.length+1;
		int expectedSum = (al*(al+1))/2;
		int actualSum = 0;
		for(int index=0; index<al-1; index++) {
			actualSum = actualSum + array[index];
		}
		System.out.println("expectedSum "+expectedSum);
		System.out.println("Missing number in given array is: "+(expectedSum-actualSum));
	}

So using one for loop you can achieve the output.

**Review by KrishnaKTechnocredits — APPROVED** [review]
Good job Nikhil. I like the way you replied to comments.

File: src/nikhilMJun20/MissingNumberFromArray.java (hunk 7 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/MissingNumberFromArray.java:17]: Yes... This is a cleaner logic.
Thanks!

**Pull request merged**